panel/lxqtpanel.cpp

```diff
@@ -1407,48 +1416,60 @@ Plugin* LXQtPanel::findPlugin(const ILXQtPanelPlugin* iPlugin) const
  ************************************************/
 QRect LXQtPanel::calculatePopupWindowPos(QPoint const & absolutePos, QSize const & windowSize) const
 {
-    int x = absolutePos.x(), y = absolutePos.y();
+    QPoint localPos = mapFromGlobal(absolutePos);
+    int x = localPos.x(), y = localPos.y();
 
     switch (position())
     {
     case ILXQtPanel::PositionTop:
-        y = mGeometry.bottom();
+        y = geometry().height();
         break;
 
     case ILXQtPanel::PositionBottom:
-        y = mGeometry.top() - windowSize.height();
+        y = 0 - windowSize.height();
         break;
 
     case ILXQtPanel::PositionLeft:
-        x = mGeometry.right();
+        x = geometry().right();
         break;
 
     case ILXQtPanel::PositionRight:
-        x = mGeometry.left() - windowSize.width();
+        x = geometry().left() - windowSize.width();
         break;
     }
 
     QRect res(QPoint(x, y), windowSize);
 
-    QRect panelScreen;
-    const auto screens = QApplication::screens();
-    if (mActualScreenNum < screens.size())
-        panelScreen = screens.at(mActualScreenNum)->geometry();
-    // NOTE: We cannot use AvailableGeometry() which returns the work area here because when in a
-    // multihead setup with different resolutions. In this case, the size of the work area is limited
-    // by the smallest monitor and may be much smaller than the current screen and we will place the
-    // menu at the wrong place. This is very bad for UX. So let's use the full size of the screen.
-    if (res.right() > panelScreen.right())
-        res.moveRight(panelScreen.right());
+    // Map to global coordinates
+    res = QRect(mapToGlobal(res.topLeft()), mapToGlobal(res.bottomRight()));
+
+    if(qGuiApp->nativeInterface<QNativeInterface::QX11Application>())
+    {
+        //On X11 we clamp rects inside screen area.
+        //NOTE: On Wayland it's done by compositor
+
+        QRect panelScreen;
+        const auto screens = QApplication::screens();
+        if (mActualScreenNum < screens.size())
+            panelScreen = screens.at(mActualScreenNum)->geometry();
+
+        // NOTE: We cannot use AvailableGeometry() which returns the work area here because when in a
+        // multihead setup with different resolutions. In this case, the size of the work area is limited
+        // by the smallest monitor and may be much smaller than the current screen and we will place the
+        // menu at the wrong place. This is very bad for UX. So let's use the full size of the screen.
 
-    if (res.bottom() > panelScreen.bottom())
-        res.moveBottom(panelScreen.bottom());
+        if (res.right() > panelScreen.right())
+            res.moveRight(panelScreen.right());
 
-    if (res.left() < panelScreen.left())
-        res.moveLeft(panelScreen.left());
+        if (res.bottom() > panelScreen.bottom())
+            res.moveBottom(panelScreen.bottom() - mGeometry.top());
```

Comment:
These checks are for preventing off-screen popups with any alignment. `mGeometry` has no role in them. Please remove it and make them as they were.